src/Targets/Microsoft.CSharp.DesignTime.targets

```diff
@@ -59,4 +112,88 @@
+    NuGet package dependencies data. It is referenced from NuGetPackageReference.xaml.
+    
+    Currently it is using ResolveNuGetPackageAssets task (which is loaded already by common imports)
+    and gets only a list of top level direct package dependencies if project.lock.file exists.
+    
+    TODO: Work with native languages team to re-write this task and/or create new tasks that would
+          read project.lock.json and get all data from it:
+            - full workld of dependencies with their corresponding types (similar to what 
+              ProjectModelServer did for existing dotnet project system).
+            - all other assets like content etc
+          Having that task there should be a first level target calling it to get data into project,
+          then this data can be transformed to other formats/item collections by next level targets
+          including ResolveNuGetPackageReferencesDesignTime target.
+  -->
+  <Target
+        Name="ResolvePackageReferencesDesignTime"
+        Returns="@(Packages)"
+        DependsOnTargets="ResolveAssemblyReferencesDesignTime"
+        Condition="exists('$(ProjectLockFile)')">
+
+    <ResolveNuGetPackageAssets AllowFallbackOnTargetSelection="$(DesignTimeBuild)"
+                               ContinueOnError="$(ContinueOnError)"
+                               IncludeFrameworkReferences="$(IncludeFrameworkReferencesFromNuGet)"
+                               NuGetPackagesDirectory="$(NuGetPackagesDirectory)"
+                               RuntimeIdentifier="$(_NuGetRuntimeIdentifierWithoutAot)"
+                               ProjectLanguage="$(Language)"
+                               ProjectLockFile="$(ProjectLockFile)"
+                               ProjectReferencesCreatingPackages="@(ProjectReferenceCreatingPackage)"
+                               ContentPreprocessorValues="@(NuGetPreprocessorValue)"
+                               ContentPreprocessorOutputDirectory="$(IntermediateOutputPath)\NuGet"
+                               TargetMonikers="$(NuGetTargetMoniker);$(_NuGetTargetFallbackMoniker)">
+
+      <Output TaskParameter="ResolvedAnalyzers" ItemName="_Analyzer" />
+      <Output TaskParameter="ResolvedCopyLocalItems" ItemName="_ReferenceCopyLocalPaths" />
+      <Output TaskParameter="ResolvedReferences" ItemName="_ReferencesFromNuGetPackages" />
+      <Output TaskParameter="ReferencedPackages" ItemName="_MyReferencedNuGetPackages" />
+      <Output TaskParameter="ContentItems" ItemName="_NuGetContentItems" />
+      <Output TaskParameter="FileWrites" ItemName="FileWrites" />
+    </ResolveNuGetPackageAssets>
+       
+    <ItemGroup>
+      <Packages Include="@(_MyReferencedNuGetPackages)"/>      
+    </ItemGroup>
+  </Target>
+  
+  <Target Name="RunLockFileToMSBuild">
+    <Message Text="     -> (Raising Lock File: ProjectLockFile=$(ProjectLockFile))" Importance="high"/>
+
+    <LockFileToMSBuild2 
```

Comment:
Yes. I commented it out for now, until Nat's task is checked in and shipped

---
In reply to: [74313364](https://github.com/dotnet/roslyn-project-system/pull/349#discussion_r74313364) [](ancestors = 74313364)